riscv/src/continuations.rs

```diff
@@ -45,15 +45,23 @@ where
 {
     let num_chunks = bootloader_inputs.len();
 
+    log::info!("Advancing pipeline to PilWithEvaluatedFixedCols stage...");
+    let pipeline = pipeline_factory();
+    let pil_with_evaluated_fixed_cols = pipeline.pil_with_evaluated_fixed_cols().unwrap();
+
+    // This returns the same pipeline as pipeline_factory() (with the same name, output dir, etc...)
+    // but starting from the PilWithEvaluatedFixedCols stage. This is more efficient, because we can advance
+    // to that stage once before we branch into different chunks.
+    let optimized_pipeline_factory = || {
+        pipeline_factory().from_pil_with_evaluated_fixed_cols(pil_with_evaluated_fixed_cols.clone())
```

Comment:
Instead of cloning, you could also change `PilWithEvaluatedFixedCols` to contain an `Rc` to the values. Same with evaluated fixed columns, they are never modified once they were created, so`Rc` seems to be perfectly suitable.

Reply:
Like this? d1a1bb2f0e0741bf46a4d7c5cac412eb777f9299